### PR TITLE
Selection highlighting, picking of MhFootPrint node, with test.

### PR DESCRIPTION
**Pull request by ppt-adsk**

### CODE REVIEW

File: lib/mayaHydra/flowViewportAPIExamples/footPrintNode/mhFootPrintNode.cpp [hunk 1]
```diff
@@ -63,6 +72,40 @@ PXR_NAMESPACE_USING_DIRECTIVE
 namespace {
 void nodeAddedToModel(MObject& node, void* clientData);
 void nodeRemovedFromModel(MObject& node, void* clientData);
+
+// Pick handler for the footprint node.
+
+class FootPrintPickHandler : public MayaHydra::PickHandler {
```
Comment (ppt-adsk) [inline lib/mayaHydra/flowViewportAPIExamples/footPrintNode/mhFootPrintNode.cpp:78]: Foot print node had no pick handler, picking did not work.

File: lib/mayaHydra/flowViewportAPIExamples/footPrintNode/mhFootPrintNode.cpp [hunk 2]
```diff
@@ -552,19 +598,30 @@ void* MhFootPrint::creator()
...
+    _pathPrefix = SdfPath(TfStringPrintf("/MhFootPrint_%p", this));
 
     //Add the callback when an attribute of this node changes
     MObject obj = thisMObject();
     _cbAttributeChangedId = MNodeMessage::addAttributeChangedCallback(obj, attributeChangedCallback, ((void*)this));
 
     //Data producer scene index interface is used to add the retained scene index to all viewports with all render delegates
     auto& dataProducerSceneIndexInterface = Fvp::DataProducerSceneIndexInterface::get();
-    dataProducerSceneIndexInterface.addDataProducerSceneIndex(_retainedSceneIndex, noPrefix, (void*)&obj, FvpViewportAPITokens->allViewports,FvpViewportAPITokens->allRenderers);
+    dataProducerSceneIndexInterface.addDataProducerSceneIndex(_retainedSceneIndex, _pathPrefix, (void*)&obj);
```
Comment (ppt-adsk) [inline lib/mayaHydra/flowViewportAPIExamples/footPrintNode/mhFootPrintNode.cpp:609]: Removed default arguments.

File: lib/mayaHydra/flowViewportAPIExamples/footPrintNode/mhFootPrintNode.cpp [hunk 3]
```diff
@@ -552,19 +598,30 @@ void* MhFootPrint::creator()
 
 void MhFootPrint::addedToModelCb()
 {
-    static const SdfPath noPrefix = SdfPath::AbsoluteRootPath();
+    _pathPrefix = SdfPath(TfStringPrintf("/MhFootPrint_%p", this));
```
Comment (ppt-adsk) [inline lib/mayaHydra/flowViewportAPIExamples/footPrintNode/mhFootPrintNode.cpp:601]: Lack of path prefix meant that the prefix was the Hydra scene index root.  Thus, when foot print node was selected, the Hydra scene index root was selected, and all its children received selection highlighting, which is the normal and correct way to highlight children of a selected prim.  Unfortunately, this meant that the whole scene was selection highlighted...  Fixed with a proper Hydra scene index prefix.

File: test/lib/mayaUsd/render/mayaToHydra/cpp/testHydraPrim.cpp [hunk 4]
```diff
@@ -108,3 +112,29 @@ TEST(TestHydraPrim, translation)
     constexpr double epsilon{1e-7};
     ASSERT_TRUE(GfIsClose(primTranslation, expectedTranslation, epsilon));
 }
+
+TEST(TestHydraPrim, countPrims)
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/cpp/testHydraPrim.cpp:116]: New utility to count Hydra scene index prims, given a regular expression.

File: test/lib/mayaUsd/render/mayaToHydra/testFootPrintNode.py [hunk 5]
```diff
@@ -257,5 +259,38 @@ def test_Load(self):
             #using imageVersion as the color is different for this image under usd 24.11+
             self.assertSnapshotClose("loadingFootPrintScene.png", self.IMAGE_DIFF_FAIL_THRESHOLD, self.IMAGE_DIFF_FAIL_PERCENT, self.imageVersion)
 
+    # Test selection highlighting.  When the footprint node is selected, only
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/render/mayaToHydra/testFootPrintNode.py:262]: Without the fixes for picking and selection highlighting, these two tests fail.

With the fixes, these two tests pass.

**Review by lanierd-adsk — APPROVED** [review]

**Pull request merged**